source/common/common/version.cc

```diff
@@ -19,18 +19,18 @@ const std::string& VersionInfo::revisionStatus() {
 }
 
 const std::string& VersionInfo::version() {
-  CONSTRUCT_ON_FIRST_USE(std::string, fmt::format("{}/{}/{}/{}/{}", revision(),
-                                                  BUILD_VERSION_NUMBER, revisionStatus(),
 #ifdef NDEBUG
-                                                  "RELEASE",
+  const std::string release_type = "RELEASE";
```

Comment:
why this change? make them string_view.

Reply:
This is another example where the macro CONSTRUCT_ON_FIRST_USE cannot have nested #cpp statements such as #ifdef.  We simply followed the file's current convention, if a rewrite is in order, doesn't it make sense to convert this all to absl::string_view? Is it possible to accept this patch as-is, to accomplish the stated purpose of this PR (to get it compiling without gcc'isms)?

Reply:
What's the gcc'ism in this function before your change? You should be able to make L23-L30 to `const absl::string_view`, basically no string allocation other than the last statement.

Reply:
@lizan we updated the file, is this what you had in mind?